### PR TITLE
Refinements fix

### PR DESCRIPTION
**Pull request by schroerbrian**
This expands the refinement limit value that we pass to the Algolia search component, if we are filtering out returned refinements – in cases where we are using `transformItems` to filter the returned refinements, we want the limit to be high to ensure that Algolia returns all of the refinements that we actually want to display. It's a little bit complicated, so I tried my best to explain it in comments. Let me know if this can be improved at all.

Tl;dr: Algolia passes back a limited amount of eligibility or category refinements. We have been filtering this returned list in some cases, without realizing that the original, unfiltered list may not even contain all of the refinements that we want to display. That's why sometimes the list on the sidebar can be very short.

### CODE REVIEW

File: app/components/ucsf/RefinementLists/ucsfEligibilitiesMap.ts [hunk 1]
```diff
@@ -22,13 +22,6 @@ interface UcsfEligibilityMap {
   [key: string]: EligibilityGroup[];
 }
 
-// N.B.: Until we have found a way to make use of Algolia's "Show More" refinements button,
```
Comment (schroerbrian) [inline app/components/ucsf/RefinementLists/ucsfEligibilitiesMap.ts:25]: Turns out this comment was moot as we are using this the eligibilities in this file to filter out those returned by Algolia anyhow

File: app/utils/whitelabel.ts [hunk 2]
```diff
@@ -45,7 +45,6 @@ interface WhiteLabelSite {
   intercom: boolean;
   logoLinkDestination: string;
   navLogoStyle: string;
-  refinementListLimit: number;
```
Comment (schroerbrian) [inline app/utils/whitelabel.ts:48]: This config value is not currently need and arguably never was. Algolia's default is 10 and in cases where we are filtering by either a static list of refinements or one returned by the API (in the case of the subcategories endpoint), we need a high value to ensure that all desired categories are returned to begin with

**Review by richardxia — APPROVED** [review]
LGTM, thanks for digging into this and understanding the issue!
